src/app/settings/settings.effects.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { TestBed } from '@angular/core/testing';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { Action } from '@ngrx/store';
+import { EffectsMetadata, getEffectsMetadata } from '@ngrx/effects';
+import { Observable } from 'rxjs';
+import { cold } from 'jasmine-marbles';
+
+import { SettingsEffects } from './settings.effects';
+import { AnimationsService, LocalStorageService } from '@app/core';
+import {
+  SettingsActionTypes,
+  SETTINGS_KEY
+} from '@app/settings/settings.reducer';
+
+describe('SettingsEffects', () => {
+  let actions$: Observable<Action>;
+  let effects: SettingsEffects;
+  let metadata: EffectsMetadata<SettingsEffects>;
+  let localStorageService: any;
+  let animationsService: any;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        SettingsEffects,
+        provideMockActions(() => actions$),
+        {
+          provide: LocalStorageService,
+          useValue: jasmine.createSpyObj('LocalStorageService', ['setItem'])
+        },
+        {
+          provide: AnimationsService,
+          useValue: jasmine.createSpyObj('AnimationsService', [
+            'updateRouteAnimationType'
+          ])
+        }
+      ]
+    });
+
+    localStorageService = TestBed.get(LocalStorageService);
+    animationsService = TestBed.get(AnimationsService);
+    effects = TestBed.get(SettingsEffects);
+  });
+
+  it('should not dispatch any action', () => {
+    metadata = getEffectsMetadata(effects);
+    expect(metadata.persistSettings).toEqual({ dispatch: false });
+  });
+
+  it('should call methods on AnimationsService and LocalStorageService for PERSIST action', () => {
+    const action = {
```

Comment:
A nitpick, do we want to create these action "manually" or would it be better to use the action creators? @tomastrajan

Reply:
@timdeschryver @hhubik I would go with the action creators so `new SomeAction()`, actions and effects are tightly coupled anyway so they can be coupled in tests too, ideas?

Reply:
Yea I agree.

Reply:
@timdeschryver changed to use action creators.